src/components/DataSourceCard/index.tsx

```diff
@@ -1,42 +1,75 @@
 import { SettingOutlined } from "@ant-design/icons";
-import { Card } from "antd";
+import { Card, Typography, Dropdown } from "antd";
 import { useTranslation } from "react-i18next";
 
 import Button from "@/components/Button";
 import DataSourceTag from "@/components/DataSourceTag";
+import formatTime, { CARD_TIME_FORMAT } from "@/utils/helpers/formatTime";
 import type { DataSourceInfo } from "@/types/dataSource";
 
 import styles from "./index.module.less";
 
 import type { FC } from "react";
 
-const DataSourceCard: FC<DataSourceInfo> = ({
+const { Paragraph } = Typography;
+
+interface DataSourceCardProps extends DataSourceInfo {
+  onEdit: (id: string | undefined) => void;
+  onDelete: (id: string | undefined) => void;
+}
+
+const DataSourceCard: FC<DataSourceCardProps> = ({
+  id,
+  name,
+  // db_params: { host },
   host,
   type,
   updatedAt,
   createdAt,
```

Comment:
this have to be `dataSource` object, you have `DataSourceInfo` type, so
`dataSource: DataSourceInfo`